### PR TITLE
[KOGITO-5680] Quarkus/Kie7 update Jenkins jobs

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-5680

Related PRs:
- https://github.com/kiegroup/kogito-pipelines/pull/233
- https://github.com/kiegroup/kogito-runtimes/pull/1514
- https://github.com/kiegroup/optaplanner/pull/1503
- https://github.com/kiegroup/kogito-examples/pull/908
- https://github.com/kiegroup/optaplanner-quickstarts/pull/188

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run all builds</b>  
  Please add comment: <b>Jenkins retest this</b>

* <b>Run (or rerun) specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] tests</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Run (or rerun) LTS specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Run (or rerun) native specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by radtriste** [comment]
Examples:
- Kie7 update PR: https://github.com/radtriste/kogito-runtimes/pull/110
- Quarkus update PR: https://github.com/radtriste/kogito-runtimes/pull/111/files

File: .jenkins/dsl/jobs.groovy [hunk 1]
```diff
@@ -94,6 +87,11 @@ if (Utils.isLTSBranch(this)) {
     setupQuarkusJob(nightlyBranchFolder, Utils.getQuarkusLTSVersion(this))
     setupNativeLTSJob(nightlyBranchFolder)
 }
+
+// Tools job
+KogitoJobTemplate.createQuarkusUpdateJob(this, 'kogito-runtimes', 'kogito', 'Kogito Runtimes')
```
Comment (Ginxo) [inline .jenkins/dsl/jobs.groovy:92]: as I said I'm not sure whether those `createQuarkusUpdateJob` and `createKie7UpdateJob` methods should be in `KogitoJobTemplate`

**Review by Ginxo — APPROVED** [review]
LGTM, just check my comments

File: .jenkins/dsl/jobs.groovy (hunk 1 above)
Comment (radtriste) [reply .jenkins/dsl/jobs.groovy:92]: you mean we should keep this in each repository instead ?
I thought about it as it would make more sense IMO

Comment (Ginxo) [reply .jenkins/dsl/jobs.groovy:92]: that's what I mean @radtriste 
we should probably keep KogitoJobTemplate as abstract as possible. We could may be create a new `KieUtils` or `JobUtils`

Comment (radtriste) [reply .jenkins/dsl/jobs.groovy:92]: nice catch. gonna update the PRs :)

**Comment by radtriste** [comment]
Examples:
- Kie7 update PR: https://github.com/radtriste/kogito-runtimes/pull/116
- Quarkus update PR: https://github.com/radtriste/kogito-runtimes/pull/115

**Comment by radtriste** [comment]
`Jenkins Tests / dsl-tests` will be ok once https://github.com/kiegroup/kogito-pipelines/pull/233 is merged

**Pull request merged**